File: 7.Go版每日一题/1646. 获取生成数组中的最大值.ts
function getMaximumGenerated(n: number): number {
	if (n === 0 || n === 1) return n
	const arr = new Array(n + 1).fill(0)
	let res = 0
	arr[1] = 1
	for (let i = 2; i <= n; i++) {
		arr[i] = arr[Math.floor(i / 2)] + (i % 2) * arr[Math.floor(i / 2) + 1]
		res = res < arr[i] ? arr[i] : res
	}
	return res
}

File: 7.Go版每日一题/1646. 获取生成数组中的最大值.go
package main

func main() {}

func getMaximumGenerated(n int) (ans int) {
	if n == 0 {
		return
	}
	arr := make([]int, n+1)
	arr[1] = 1
	ans = 1 // start from 1
	for i := 2; i <= n; i++ {
		arr[i] = arr[i/2] + i%2*arr[(i/2)+1]
		ans = myMax(ans, arr[i])
	}
	return
}

func myMax(a, b int) int {
	if a > b {
		return a
	}
	return b
}
